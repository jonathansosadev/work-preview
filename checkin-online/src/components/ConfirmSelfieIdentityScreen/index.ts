export {ConfirmSelfieIdentityScreen as default} from './ConfirmSelfieIdentityScreen';
