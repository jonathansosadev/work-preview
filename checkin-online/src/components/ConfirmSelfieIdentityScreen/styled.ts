import styled from 'styled-components';
import {device} from '../../styled/device';
import Webcam from '../Webcam';
import {StyledWebcam} from '../Webcam/styled';

export const Content = styled.div`
  text-align: center;
  cursor: default;
`;

export const Title = styled.div`
  font-family: ProximaNova-Bold, sans-serif;
  font-size: 16px;
  max-width: 311px;
  margin: -40px auto 0;
  text-transform: uppercase;
  color: #161643;
`;

export const Subtitle = styled.div`
  font-family: ProximaNova-Light, sans-serif;
  font-size: 18px;
  max-width: 400px;
  margin: 0 auto 30px;
  color: #161643;

  @media (max-width: ${device.tablet}) {
    max-width: 311px;
  }
`;

export const SelfieWebcam = styled(Webcam)`
  height: 240px;
  border-radius: 0;

  & ${StyledWebcam} {
    height: 290px;

    @media (max-width: ${device.tablet}) {
      min-height: 220%;
    }
  }
`;

export const SelfieArea = styled.div`
  background-color: transparent;
  border: 3px solid #1a8cff;
  width: 195px;
  height: 195px;
  box-sizing: border-box;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 2;
  margin: auto;
`;

export const ButtonWrapper = styled.div`
  margin: 0px 0 35px;
  display: flex;
  justify-content: center;
`;
