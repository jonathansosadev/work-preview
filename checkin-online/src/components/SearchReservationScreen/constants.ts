enum SEARCH_RESERVATION_FORM_NAMES {
  booking_reference = 'booking_reference',
  lead_email = 'default_invite_email',
  check_in_date = 'check_in_date',
  check_out_date = 'check_out_date',
}

enum MOBILE_SEARCH_STEPS {
  bookingReference = 1,
  dateAndEmail = 2,
}

export {SEARCH_RESERVATION_FORM_NAMES, MOBILE_SEARCH_STEPS};
