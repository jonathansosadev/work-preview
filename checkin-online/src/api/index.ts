import i18next from 'i18next';
import {isMobile} from 'react-device-detect';
import * as Sentry from '@sentry/react';
import {refreshFetch} from '../utils/refreshFetch';
import {getAndHandleResponseError, getResponseHeaders} from '../utils/api';
import * as reservations from './reservations';
import * as locations from './locations';
import * as auth from './auth';
import * as purposesOfStay from './purposesOfStay';
import * as documents from './documents';
import * as guests from './guests';
import * as ocr from './ocr';
import * as statTaxExemptions from './statTaxExemptions';
import * as testErrors from './testErrors';
import * as alice from './alice';
import * as seasons from './seasons';
import * as guestPaymentAccounts from './guestPaymentAccounts';
import * as guestPayments from './guestPayments';
import * as paymentsSettings from './paymentsSettings';
import * as seasonGuests from './seasonGuests';
import * as securityDeposits from './securityDeposits';
import * as guestForms from './guestForms';
import * as reservationPayments from './reservationPayments';
import * as housingExemptions from './housingExemptions';
import * as locks from './locks';
import * as getCountryByIp from './getCountryByIp';
import * as universalLink from './universalLink';
import * as validators from './validators';
import * as upselling from './upselling';

const USER_TOKEN = 'user';
const BASE_URL = process.env.REACT_APP_BASE_URL;
const RETRY_RESPONSE_STATUSES = [500, 503, 504];
const MAX_RETRIES_NUMBER = 1;

function getURL(endpoint = '') {
  return `${BASE_URL}/api/v3/${endpoint}`;
}

export type ResolverTypes<T = any> = {
  data: T;
  error: any;
  aborted?: boolean;
};

type ResolverData = {
  customURL?: string;
  retries?: number;
};

async function resolver<T = any>(
  endpoint: string,
  options?: any,
  data?: ResolverData,
): Promise<ResolverTypes<T>> {
  const result: ResolverTypes = {
    data: null,
    error: null,
    aborted: false,
  };

  try {
    const {response, body} = await refreshFetch(data?.customURL || getURL(endpoint), {
      headers: getAuthAndBaseHeaders(),
      ...options,
    });

    if (options?.signal?.aborted) {
      console.warn('ABORTED');
      return result;
    }

    if (response.status === 204) {
      result.data = 'success';
      return result;
    }

    const responseError = getAndHandleResponseError(response, body, options);
    if (responseError) {
      result.error = responseError;
      return result;
    }

    result.data = body;
  } catch (error) {
    if (options?.signal?.aborted) {
      result.aborted = true;
    }

    const retries = data?.retries || 0;
    if (
      RETRY_RESPONSE_STATUSES.includes(error?.response?.status) &&
      retries < MAX_RETRIES_NUMBER
    ) {
      return resolver(endpoint, options, {
        ...data,
        retries: retries + 1,
      });
    }

    const responseError = getAndHandleResponseError(
      error?.response,
      error?.body,
      options,
    );
    if (responseError) {
      result.error = responseError;
      return result;
    } else {
      const headers = getResponseHeaders(error?.response);

      Sentry.withScope(scope => {
        scope.setFingerprint([
          data?.customURL || getURL(endpoint),
          String(error?.response?.statusCode),
        ]);
        Sentry.captureException(error);
        Sentry.captureMessage(`Headers: ${headers || 'no headers'}`);
      });
    }

    result.error = error;
  }

  return result;
}

type AuthTypes = {
  [key: string]: string;
};

function getAuthAndBaseHeaders(): AuthTypes {
  return {
    Authorization: `JWT ${getTokenFromLocalStorage()}`,
    'Content-Type': 'application/json',
    'Accept-Language': i18next.language.slice(0, 2),
    'X-source': isMobile ? 'M' : 'D',
  };
}

function getAnonymousHeaders() {
  return {
    'Content-Type': 'application/json',
  };
}

function getTokenFromLocalStorage(): string | null {
  let token = null;

  try {
    const rawUser = localStorage.getItem(USER_TOKEN);
    if (rawUser !== null) {
      const user = JSON.parse(rawUser);
      token = user.token;
    }
  } catch (err) {
    console.error(err);
  }

  return token;
}

function persistTokenToLocalStorage(token = ''): void {
  localStorage.setItem(USER_TOKEN, JSON.stringify({token}));
}

function removeTokenFromLocalStorage(): void {
  localStorage.removeItem(USER_TOKEN);
}

const api = {
  reservations,
  locations,
  auth,
  purposesOfStay,
  documents,
  guests,
  ocr,
  statTaxExemptions,
  testErrors,
  alice,
  seasons,
  guestPaymentAccounts,
  guestPayments,
  paymentsSettings,
  seasonGuests,
  securityDeposits,
  guestForms,
  reservationPayments,
  housingExemptions,
  locks,
  getCountryByIp,
  universalLink,
  validators,
  upselling,
};

export default api;

export {
  getURL,
  resolver,
  getAuthAndBaseHeaders,
  getAnonymousHeaders,
  getTokenFromLocalStorage,
  persistTokenToLocalStorage,
  removeTokenFromLocalStorage,
};
