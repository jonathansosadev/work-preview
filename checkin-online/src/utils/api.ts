import * as Sentry from '@sentry/react';
import {getAuthAndBaseHeaders} from '../api';

const FAILURE_RESPONSE_STATUSES = [400, 401, 429];

function stringifySentryRequestInit(options: any) {
  const base64Fields = ['signature', 'picture_file', 'front_side_scan', 'back_side_scan'];
  const body = options?.body;

  if (body) {
    const parsedBody = JSON.parse(body);

    base64Fields.forEach(field => {
      if (parsedBody[field]) {
        parsedBody[field] = 'base64';
      }
    });

    return JSON.stringify({
      ...options,
      body: parsedBody,
    });
  }

  return JSON.stringify(options);
}

function getResponseHeaders(response?: Response) {
  let headers = '';

  if (!response) {
    return headers;
  }

  // @ts-ignore
  for (const pair of response?.headers?.entries()) {
    headers += `${pair[0]}: ${pair[1]}; `;
  }
  return headers;
}

function getAndHandleResponseError(response?: Response, body?: any, options?: any) {
  if (!response) {
    return null;
  }

  if (FAILURE_RESPONSE_STATUSES.includes(response.status)) {
    return body;
  }

  if (response.status === 401) {
    Sentry.captureMessage(
      `Error 401: ${response.url}. Headers: ${JSON.stringify(
        getAuthAndBaseHeaders(),
      )}. Request init: ${stringifySentryRequestInit(options)}`,
    );
  }

  if (response.status === 404) {
    Sentry.captureMessage(
      `Error 404: ${response.url}. Request init: ${stringifySentryRequestInit(options)}`,
    );

    return {
      message: `404 Not found. Please contact support.`,
      body,
    };
  }

  if (response.status === 405) {
    Sentry.captureMessage(
      `Error 405: ${response.url}. Request init: ${stringifySentryRequestInit(options)}`,
    );

    return {
      message: `405 Method not allowed. Please contact support.`,
      body,
    };
  }

  if (response.status === 500) {
    Sentry.captureMessage(
      `Error 500: ${response.url}. Request init: ${stringifySentryRequestInit(options)}`,
    );

    return {
      message: '500 Internal Server error. Please contact support.',
      body,
    };
  }

  if (response.status === 502) {
    Sentry.captureMessage(
      `Error 502: ${response.url}. Request init: ${stringifySentryRequestInit(options)}`,
    );

    return {
      message: '502 Bad Gateway. Please contact support.',
      body,
    };
  }

  if (response.status === 503) {
    Sentry.captureMessage(
      `Error 503: ${response.url}. Request init: ${stringifySentryRequestInit(options)}`,
    );

    return {
      message: '503 Service Unavailable. Please contact support.',
      body,
    };
  }

  if (response.status === 504) {
    Sentry.captureMessage(
      `Error 504: ${response.url}. Request init: ${stringifySentryRequestInit(options)}`,
    );

    return {
      message: '504 Gateway Timeout. Please contact support.',
      body,
    };
  }

  return null;
}

export {getAndHandleResponseError, getResponseHeaders};
