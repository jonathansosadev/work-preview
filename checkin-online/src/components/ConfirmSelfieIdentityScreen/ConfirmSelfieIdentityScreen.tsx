import React from 'react';
import {useTranslation} from 'react-i18next';
import {useHistory, useLocation} from 'react-router-dom';
import api from '../../api';
import {useComputedReservationDetails} from '../../context/computedReservationDetails';
import {
  useAbortController,
  useErrorModal,
  useIsMounted,
  useModalControls,
} from '../../utils/hooks';
import {OCR_CHECKING_STATUSES} from '../../utils/constants';
import {WebcamRefTypes} from '../Webcam/Webcam';
import {getIdentityVerificationHeaderProps} from '../ConfirmIdentityGuideScreen';
import femaleIcon from '../../assets/female.svg';
import scanningIcon from '../../assets/scanning_icn.svg';
import scanningErrorIcon from '../../assets/scanning_doc_error.svg';
import Header from '../Header';
import Button from '../Button';
import Modal from '../Modal';
import {ModalSecondaryButton} from '../../styled/common';
import {
  ButtonWrapper,
  Content,
  SelfieArea,
  SelfieWebcam,
  Subtitle,
  Title,
} from './styled';

const VIDEO_CONSTRAINTS = {
  facingMode: 'user',
};

type LocationTypes = {
  documentCheckId: string;
  documentPhoto: string;
  isVerifying: boolean;
  failedOCRAttempts?: number;
};

function ConfirmSelfieIdentityScreen() {
  const {t} = useTranslation();
  const isMounted = useIsMounted();
  const history = useHistory();
  const location = useLocation<LocationTypes>();
  const {ErrorModal, displayError} = useErrorModal();
  const reservationDetails = useComputedReservationDetails();
  const webcamRef = React.useRef<WebcamRefTypes>(null);
  const [isCaptureButtonDisabled, setIsCaptureButtonDisabled] = React.useState(true);

  const {abortController, setupAbortController} = useAbortController();
  const {
    isOpen: isDetectingModalOpen,
    openModal: openDetectingModal,
    closeModal: closeDetectingModal,
  } = useModalControls();
  const {
    isOpen: isComparingModalOpen,
    openModal: openComparingModal,
    closeModal: closeComparingModal,
  } = useModalControls();

  const areBiomatchRetriesEnabled =
    reservationDetails.reservation.are_biomatch_retries_enabled;

  const goBack = React.useCallback(() => {
    history.push('/verification/document', location.state);
  }, [history, location.state]);

  React.useEffect(() => {
    const hasPersistedState =
      location.state?.documentCheckId && location.state?.documentPhoto;

    if (!hasPersistedState) {
      goBack();
    }
  }, [goBack, location.state]);

  const handleError = (error?: any, aborted?: boolean) => {
    closeDetectingModal();
    closeComparingModal();
    if (!aborted) {
      displayError(error);
    }
  };

  const getPersistedState = (result: string, selfiePhoto: string) => {
    return {
      ...location.state,
      selfiePhoto,
      facesComparingResult: result,
    };
  };

  const handleOCRMatchingCompletion = (data: any, image: string) => {
    let persistedState = getPersistedState(data, image);
    closeComparingModal();
    if (!data?.is_match) {
      const failedAttempts = persistedState.failedOCRAttempts || 0;
      persistedState = {
        ...persistedState,
        failedOCRAttempts: areBiomatchRetriesEnabled ? failedAttempts + 1 : 0,
      };
    }
    history.push('result', persistedState);
  };

  const abortCurrentRequest = () => {
    abortController.abort();
  };

  const enableScreenshotButton = () => {
    if (isMounted.current) {
      setIsCaptureButtonDisabled(false);
    }
  };

  const checkSelfieAndDocumentFacesComparing = async (id: string, image: string) => {
    const signal = setupAbortController();
    const {
      data,
      error,
      aborted,
    } = await api.ocr.checkSelfieAndDocumentFacesComparingStatus(id, {
      signal,
    });
    const isPhotoVerificationDisabled = Boolean(
      reservationDetails.reservation?.is_biomatch_photo_verification_disabled,
    );
    const isInProgress =
      !isPhotoVerificationDisabled && data?.status === OCR_CHECKING_STATUSES.processing;
    const isComplete =
      isPhotoVerificationDisabled || data?.status === OCR_CHECKING_STATUSES.complete;
    const isError = data?.status === OCR_CHECKING_STATUSES.error;

    if (!isMounted.current) {
      return;
    }

    if (error) {
      handleError(error, aborted);
      return;
    }

    if (isInProgress) {
      await checkSelfieAndDocumentFacesComparing(id, image);
      return;
    }

    if (isComplete) {
      handleOCRMatchingCompletion(data, image);
      return;
    }

    if (isError) {
      handleError();
    }
  };

  const startSelfieAndDocumentFacesComparing = async (
    selfieCheckId = '',
    image: string,
  ) => {
    openComparingModal();

    const signal = setupAbortController();
    const {data, error, aborted} = await api.ocr.compareSelfieAndDocumentFaces({
      signal,
      selfieCheckId,
      documentCheckId: location.state?.documentCheckId!,
      reservationId: reservationDetails?.reservation?.id,
    });

    if (!isMounted.current) {
      return;
    }

    if (data) {
      await checkSelfieAndDocumentFacesComparing(data?.id, image);
    }
    if (error) {
      handleError(error, aborted);
    }
  };

  const checkSelfieFaceDetection = async (id: string, image: string) => {
    const signal = setupAbortController();
    const {data, error, aborted} = await api.ocr.checkSelfieFaceDetection(id, {
      signal,
    });

    const isPhotoVerificationDisabled = Boolean(
      reservationDetails.reservation?.is_biomatch_photo_verification_disabled,
    );
    const isInProgress =
      !isPhotoVerificationDisabled && data?.status === OCR_CHECKING_STATUSES.processing;
    const isComplete =
      isPhotoVerificationDisabled ||
      (data?.status === OCR_CHECKING_STATUSES.complete && data?.is_face_detected);
    const isError =
      data.status === OCR_CHECKING_STATUSES.error || data?.is_face_detected === false;

    if (!isMounted.current) {
      return;
    }

    if (error) {
      handleError(error, aborted);
      return;
    }

    if (isInProgress) {
      await checkSelfieFaceDetection(id, image);
      return;
    }

    if (isComplete) {
      closeDetectingModal();
      await startSelfieAndDocumentFacesComparing(id, image);
      return;
    }

    if (isError) {
      handleError();
    }
  };

  const startSelfieFaceDetection = async (image = '', reservationId = '') => {
    openDetectingModal();

    const signal = setupAbortController();
    const {data, error, aborted} = await api.ocr.sendSelfieForFaceDetection(
      image,
      reservationId,
      {
        signal,
      },
    );

    if (!isMounted.current) {
      return;
    }

    if (error) {
      handleError(error, aborted);
    }
    if (data) {
      await checkSelfieFaceDetection(data.id, image);
    }
  };

  const captureScreenshot = async () => {
    try {
      const image = webcamRef?.current?.getScreenshot();
      const reservationId = reservationDetails?.reservation?.id;
      await startSelfieFaceDetection(image, reservationId);
    } catch (err) {
      displayError(err);
    }
  };

  return (
    <>
      <ErrorModal iconSrc={scanningErrorIcon} />
      <Modal
        open={isComparingModalOpen}
        title={t('comparing')}
        text={t('we_are_comparing_doc_and_selfie')}
        iconSrc={scanningIcon}
        iconAlt="Scanning female"
      >
        <ModalSecondaryButton onClick={abortCurrentRequest}>
          {t('cancel')}
        </ModalSecondaryButton>
      </Modal>
      <Modal
        open={isDetectingModalOpen}
        title={t('detecting')}
        text={t('we_are_detecting_your_face')}
        iconSrc={femaleIcon}
        iconAlt="Female"
      >
        <ModalSecondaryButton onClick={abortCurrentRequest}>
          {t('cancel')}
        </ModalSecondaryButton>
      </Modal>
      <Header
        {...getIdentityVerificationHeaderProps(reservationDetails)}
        hideBackButton={isDetectingModalOpen || isComparingModalOpen}
        title={t('verify_your_identity')}
        onBack={goBack}
      />
      <Content>
        <Title>{t('take_a_selfie')}</Title>
        <Subtitle>{t('take_a_selfie_tips')}</Subtitle>
        <SelfieWebcam
          hideScanArea
          mirrored
          ref={webcamRef}
          videoConstraints={VIDEO_CONSTRAINTS}
          onUserMedia={enableScreenshotButton}
        >
          <SelfieArea />
        </SelfieWebcam>
        <ButtonWrapper>
          <Button
            label={t('capture')}
            disabled={isCaptureButtonDisabled}
            onClick={captureScreenshot}
          />
        </ButtonWrapper>
      </Content>
    </>
  );
}

export {ConfirmSelfieIdentityScreen};
