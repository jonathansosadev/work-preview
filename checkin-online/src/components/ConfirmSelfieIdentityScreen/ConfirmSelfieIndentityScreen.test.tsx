import React from 'react';
import api from '../../api';
import {fireEvent} from '@testing-library/react';
import {renderWithProviders} from '../../utils/test';
import {OCR_CHECKING_STATUSES} from '../../utils/constants';
import {ConfirmSelfieIdentityScreen} from './ConfirmSelfieIdentityScreen';

jest.mock('../../api');

test('takes photo, identifies face on it and compares two photos', async () => {
  (api.ocr.sendSelfieForFaceDetection as jest.Mock).mockResolvedValue({
    error: {
      message: 'send selfie error',
    },
  });
  (api.ocr.checkSelfieFaceDetection as jest.Mock).mockResolvedValue({
    error: {
      message: 'check selfie error',
    },
  });
  (api.ocr.compareSelfieAndDocumentFaces as jest.Mock).mockResolvedValue({
    error: {
      message: 'compare docs error',
    },
  });
  (api.ocr.checkSelfieAndDocumentFacesComparingStatus as jest.Mock).mockResolvedValue({
    error: {
      message: 'check compare docs error',
    },
  });
  (api.housingExemptions.getById as jest.Mock).mockResolvedValue({
    data: null,
  });

  const {
    getByText,
    findByAltText,
    findByText,
    queryByAltText,
    queryByText,
    history,
  } = renderWithProviders(<ConfirmSelfieIdentityScreen />);
  const historySpy = jest.spyOn(history, 'push');

  expect(queryByAltText(/exclamation error mark/i)).not.toBeInTheDocument();

  const captureButton = getByText(/^capture$/);
  fireEvent.click(captureButton);
  expect(await findByText(/send selfie error/)).toBeInTheDocument();

  const selfieCheckId = 'testId';
  (api.ocr.sendSelfieForFaceDetection as jest.Mock).mockResolvedValue({
    data: {
      id: selfieCheckId,
    },
  });
  fireEvent.click(captureButton);
  expect(await findByText(/check selfie error/)).toBeInTheDocument();
  expect(queryByText(/send selfie error/)).not.toBeInTheDocument();
  expect((api.ocr.checkSelfieFaceDetection as jest.Mock).mock.calls[0][0]).toEqual(
    selfieCheckId,
  );
  let closeErrorModalButton = getByText(/ok/i);
  fireEvent.click(closeErrorModalButton);
  expect(queryByText(/check selfie error/)).not.toBeInTheDocument();

  (api.ocr.checkSelfieFaceDetection as jest.Mock).mockResolvedValue({
    error: null,
    data: {
      status: OCR_CHECKING_STATUSES.error,
    },
  });
  fireEvent.click(captureButton);
  let detectingModal = queryByAltText(/female/i);
  expect(detectingModal).toBeInTheDocument();
  let errorModal = await findByAltText(/exclamation error mark/i);
  expect(errorModal).toBeInTheDocument();
  expect((api.ocr.checkSelfieFaceDetection as jest.Mock).mock.calls[1][0]).toEqual(
    selfieCheckId,
  );
  closeErrorModalButton = getByText(/ok/i);
  fireEvent.click(closeErrorModalButton);
  expect(queryByAltText(/exclamation error mark/i)).not.toBeInTheDocument();

  (api.ocr.checkSelfieFaceDetection as jest.Mock).mockResolvedValue({
    error: null,
    data: {
      status: OCR_CHECKING_STATUSES.complete,
      is_face_detected: false,
    },
  });
  fireEvent.click(captureButton);
  detectingModal = queryByAltText(/female/i);
  expect(detectingModal).toBeInTheDocument();
  errorModal = await findByAltText(/exclamation error mark/i);
  expect(errorModal).toBeInTheDocument();
  expect((api.ocr.checkSelfieFaceDetection as jest.Mock).mock.calls[2][0]).toEqual(
    selfieCheckId,
  );
  closeErrorModalButton = getByText(/ok/i);
  fireEvent.click(closeErrorModalButton);
  expect(queryByAltText(/exclamation error mark/i)).not.toBeInTheDocument();

  (api.ocr.checkSelfieFaceDetection as jest.Mock).mockResolvedValue({
    error: null,
    data: {
      status: OCR_CHECKING_STATUSES.complete,
      is_face_detected: true,
    },
  });
  fireEvent.click(captureButton);
  detectingModal = queryByAltText(/female/i);
  expect(detectingModal).toBeInTheDocument();
  expect(await findByText(/compare docs error/)).toBeInTheDocument();
  expect((api.ocr.checkSelfieFaceDetection as jest.Mock).mock.calls[3][0]).toEqual(
    selfieCheckId,
  );

  const compareSelfieAndDocId = 'testCompareId';
  (api.ocr.compareSelfieAndDocumentFaces as jest.Mock).mockResolvedValue({
    error: null,
    data: {
      id: compareSelfieAndDocId,
    },
  });
  fireEvent.click(captureButton);
  detectingModal = queryByAltText(/female/i);
  expect(detectingModal).toBeInTheDocument();
  errorModal = await findByText(/check compare docs error/i);
  expect(errorModal).toBeInTheDocument();
  expect((api.ocr.checkSelfieFaceDetection as jest.Mock).mock.calls[4][0]).toEqual(
    selfieCheckId,
  );
  expect(
    (api.ocr.checkSelfieAndDocumentFacesComparingStatus as jest.Mock).mock.calls[0][0],
  ).toEqual(compareSelfieAndDocId);
  closeErrorModalButton = getByText(/ok/i);
  fireEvent.click(closeErrorModalButton);
  expect(queryByText(/check compare docs error/i)).not.toBeInTheDocument();

  (api.ocr.checkSelfieAndDocumentFacesComparingStatus as jest.Mock).mockResolvedValue({
    error: null,
    data: {
      status: OCR_CHECKING_STATUSES.error,
      is_match: false,
    },
  });
  fireEvent.click(captureButton);
  detectingModal = queryByAltText(/female/i);
  expect(detectingModal).toBeInTheDocument();
  errorModal = await findByAltText(/exclamation error mark/i);
  expect(errorModal).toBeInTheDocument();
  expect((api.ocr.checkSelfieFaceDetection as jest.Mock).mock.calls[5][0]).toEqual(
    selfieCheckId,
  );
  expect(
    (api.ocr.checkSelfieAndDocumentFacesComparingStatus as jest.Mock).mock.calls[1][0],
  ).toEqual(compareSelfieAndDocId);

  (api.ocr.checkSelfieAndDocumentFacesComparingStatus as jest.Mock).mockResolvedValue({
    error: null,
    data: {
      status: OCR_CHECKING_STATUSES.complete,
      is_match: true,
    },
  });
  fireEvent.click(captureButton);
  detectingModal = queryByAltText(/female/i);
  expect(detectingModal).toBeInTheDocument();
  await findByAltText(/scanning female/i);
  expect(historySpy.mock.calls[0]).toMatchSnapshot();
});
