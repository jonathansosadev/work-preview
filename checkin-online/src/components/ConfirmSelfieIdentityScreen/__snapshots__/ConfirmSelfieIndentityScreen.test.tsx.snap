// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`takes photo, identifies face on it and compares two photos 1`] = `
Array [
  "facesComparingResult",
  Object {
    "facesComparingResult": Object {
      "is_match": true,
      "status": "COM",
    },
    "selfiePhoto": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQYAAACaCAYAAABL5uawAAAHaUlEQVR4Xu3cP4ylZRUH4PPeO4s1CZCYUJA47hJiJZWVS2cBGgu1IksI7C4hVFphATRWSCWBWQRpTEy2MBoTjRVUhIKKAO4fEgoDha2N7sz3kpmNcVk2vEs4O9w57zPl5t1zv99zvvxy79x7p4UfAgQIXCfQiBAgQOB6AcXgniBA4HMCisFNQYCAYnAPECAwFvCMYWzkBIHpBBTDdCsXmMBYQDGMjZwgMJ2AYphu5QITGAsohrGREwSmE1AM061cYAJjAcUwNnKCwHQCimG6lQtMYCygGMZGThCYTkAxTLdygQmMBRTD2MgJAtMJKIbpVi4wgbGAYhgbOUFgOgHFMN3KBSYwFlAMYyMnCEwnoBimW7nABMYCimFs5ASB6QQUw3QrF5jAWEAxjI2cIDCdgGKYbuUCExgLKIaxkRMEphNQDNOtXGACYwHFMDZygsB0AophupULTGAsoBjGRk4QmE5AMUy3coEJjAUUw9jICQLTCSiG6VYuMIGxgGIYGzlBYDoBxTDdygUmMBZQDGMjJwhMJ6AYplu5wATGAophbOQEgekEFMN0KxeYwFhAMYyNnCAwnYBimG7lAhMYCyiGsZETBKYTUAzTrVxgAmMBxTA2coLAdAKKYbqVC0xgLKAYxkZOEJhOQDFMt3KBCYwFFMPYyAkC0wkohulWLjCBsYBiGBs5QWA6AcUw3coFJjAWUAxjIycITCegGKZbucDb5658f7W3fLOv1/9et+WTfzx+2ztUPiugGNwR5QW2X7pychXLybZaf7dHf+hGgXuLN/b//dLpYw+UB7mJgIrhJpAcOZoCJ87t/nDpy49atEdvmKC1f0WPP0b0H/cW7ymG/ysphqN5z7vqLxDYf4bQ1vFM63Hy+mMt2t/7svfWEqs3Lj9x7OBZgp/PCygGd0UpgeMv/efZWK1PRfR7/hds/2VC21veVAY3v2rFcPNWTm6wwPbL/b622n2x9fhORNwRLd7rsdrpe8u7nhl8+cUphi9v5n9smMDxnd0/RPSfRWsfx9I/bKu2c+H01u837DKP1OUohiO1Lhd7rcDB7xJaPN9a3B8Rf4tlefviE994ltJXF1AMX93QhEMWOHj7cd1ORe+PHDx0a6+v+/rpD860Tw75Uso+nGIou9qawa795WKLeGtviaf9DiF/14oh39TEWyRwtRRWzxyMX5bnvMtwi6D3n4TdutEmE8gROPHylR/01h6M6E9GtI96i6cund76S850U24koBjcFxstcO1Lhx79/FY/9ssPzrZLG33RBS5OMRRYYsUI3/pt314tu6+0Je6LFne13h67cHbr1YpZNzGTYtjErUx+Tcd3rjwfET+/+rKhf+SLTYd/QyiGwzf3iDcQ2H6l371e9r7XY/lVRNvef9nQ2up3F09v/RXY4QsohsM394jXCVz7bsN+Iaza6k8+ufj13iaK4ev1n/rR7z3Xj/e++3qP+PbB9xsifn3xzLFfTI2yIeEVw4YsYrbLuPrpxfhNRLu9R7/Y29bDlx9v/5zNYVPzKoZN3Uzh6zqxs/uTHv2FiLh7/4NKvt+wectWDJu3k9JX9NmPNLefXjizdb504CMaTjEc0cUd1cs+sbP75/2/u7gs8YDvOGzuFhXD5u6m5JVtv9bvjP/GnZfPtvdLBiwSSjEUWaQYBDIFFEOmplkEiggohiKLFINApoBiyNQ0i0ARAcVQZJFiEMgUUAyZmmYRKCKgGIosUgwCmQKKIVPTLAJFBBRDkUWKQSBTQDFkappFoIiAYiiySDEIZAoohkxNswgUEVAMRRYpBoFMAcWQqWkWgSICiqHIIsUgkCmgGDI1zSJQREAxFFmkGAQyBRRDpqZZBIoIKIYiixSDQKaAYsjUNItAEQHFUGSRYhDIFFAMmZpmESgioBiKLFIMApkCiiFT0ywCRQQUQ5FFikEgU0AxZGqaRaCIgGIoskgxCGQKKIZMTbMIFBFQDEUWKQaBTAHFkKlpFoEiAoqhyCLFIJApoBgyNc0iUERAMRRZpBgEMgUUQ6amWQSKCCiGIosUg0CmgGLI1DSLQBEBxVBkkWIQyBRQDJmaZhEoIqAYiixSDAKZAoohU9MsAkUEFEORRYpBIFNAMWRqmkWgiIBiKLJIMQhkCiiGTE2zCBQRUAxFFikGgUwBxZCpaRaBIgKKocgixSCQKaAYMjXNIlBEQDEUWaQYBDIFFEOmplkEiggohiKLFINApoBiyNQ0i0ARAcVQZJFiEMgUUAyZmmYRKCKgGIosUgwCmQKKIVPTLAJFBBRDkUWKQSBTQDFkappFoIiAYiiySDEIZAoohkxNswgUEVAMRRYpBoFMAcWQqWkWgSICiqHIIsUgkCmgGDI1zSJQREAxFFmkGAQyBRRDpqZZBIoIKIYiixSDQKaAYsjUNItAEQHFUGSRYhDIFFAMmZpmESgioBiKLFIMApkCiiFT0ywCRQQUQ5FFikEgU0AxZGqaRaCIgGIoskgxCGQKKIZMTbMIFBFQDEUWKQaBTAHFkKlpFoEiAoqhyCLFIJAp8CkAmtmbi7vs2AAAAABJRU5ErkJggg==",
  },
]
`;
